lib/keystores.js

```diff
@@ -256,8 +256,12 @@ exports.getStorageUsage = async ({
     // if custom aggregator has been given, call it
     if(aggregate) {
       counters.push(limit(() => {
-        _checkComputeStorageSignal({signal, meterId});
-        return aggregate({meterId, config, usage});
+        try {
+          _checkComputeStorageSignal({signal, meterId});
+          return aggregate({meterId, config, usage});
+        } catch(e) {
+          return e;
```

Comment:
Returning an error here does not look right. I don't understand these changes. What's the problem they are trying to solve?

Reply:
So the way it was set up before it would properly throw the error however it would also have an "UnhandledPromiseRejectionWarning". This was related to us throwing an error inside of the limit() function. So in order to resolve that "UnhandledPromiseRejectionWarning" error, we have to return the error instead. The end result is still a thrown error like expected.

Reply:
I think there must be some weirdness with the promise fun library return values.  I have observed some strange behavior in logs around UnhandledPromiseRejectionWarning when p-retry is used.

Reply:
@JoshDunnDev, make `_checkComputeStorageSignal` async and see if it fixes it instead. It won't be converted into a rejected promise without that. So add `async` to the helper function.